src/System.Memory/src/System/Buffers/ReadOnlySequence_helpers.cs

```diff
@@ -11,207 +12,171 @@ namespace System.Buffers
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static SequencePosition SeekMultiSegment(IMemoryList<byte> start, int startIndex, IMemoryList<byte> end, int endPosition, long count, bool checkEndReachable)
+        private static SequencePosition SeekMultiSegment(object startSegment, int startIndex, object endSegment, int endPosition, long count, bool checkEndReachable)
         {
+            var current = (IMemoryList<T>)startSegment;
+            var end = (IMemoryList<T>)endSegment;
+
             SequencePosition result = default;
-            bool foundResult = false;
-            IMemoryList<byte> current = start;
             int currentIndex = startIndex;
 
-            while (current != null)
+            do
             {
-                // We need to loop up until the end to make sure start and end are connected
-                // if end is not trusted
-                if (!foundResult)
-                {
-                    var isEnd = current == end;
-                    int currentEnd = isEnd ? endPosition : current.Memory.Length;
-                    int currentLength = currentEnd - currentIndex;
-
-                    // We would prefer to put position in the beginning of next segment
-                    // then past the end of previous one, but only if we are not leaving current buffer
-                    if (currentLength > count ||
-                       (currentLength == count && isEnd))
-                    {
-                        result = new SequencePosition(current, currentIndex + (int)count);
-                        foundResult = true;
-                        if (!checkEndReachable)
-                        {
-                            break;
-                        }
-                    }
-
-                    count -= currentLength;
-                }
+                bool isCurrentAtEnd = current == end;
+
+                int memoryLength = isCurrentAtEnd ? endPosition - currentIndex : current.Memory.Length - currentIndex;
 
-                if (current.Next == null && current != end)
+                if (memoryLength > count || (memoryLength == count && isCurrentAtEnd))
                 {
-                    ThrowHelper.ThrowInvalidOperationException_EndPositionNotReached();
+                    result = new SequencePosition(current, currentIndex + (int)count);
+                    break;
                 }
 
-                current = current.Next;
+                count -= memoryLength;
                 currentIndex = 0;
-            }
+                current = current.Next;
+            } while (current != null);
 
-            if (!foundResult)
-            {
+            if (current == null)
                 ThrowHelper.ThrowArgumentOutOfRangeException_CountOutOfRange();
+
+            if (checkEndReachable)
+            {
+                while (current.Next != null)
+                    current = current.Next;
```

Comment:
Why do you hate braces? 😄

Reply:
I'm adaptable 😄

Reply:
I think my VS has been a bit weird ever since I edited some files in mono